recorder.go

```diff
@@ -188,20 +207,39 @@ type Recorder struct {
 	// collection requests.
 	accessToken string
 
-	tracerID uint64
+	tracerID           uint64        // the LightStep tracer guid
+	verbose            bool          // whether to print verbose messages
+	maxReportingPeriod time.Duration // set by Options.MaxReportingPeriod
+	reportingTimeout   time.Duration // set by Options.ReportTimeout
+
+	// Remote service that will receive reports.
+	backend cpb.CollectorServiceClient
+
+	//////////////////////////////////////////////////////////
+	// MUTABLE MUTABLE MUTABLE MUTABLE MUTABLE MUTABLE MUTABLE
+	//////////////////////////////////////////////////////////
 
-	verbose bool
+	// Two buffers of data.
+	buffer   spansBuffer
+	flushing spansBuffer
+
+	// Flush state.
+	reportInFlight    bool
+	lastReportAttempt time.Time
 
 	// We allow our remote peer to disable this instrumentation at any
 	// time, turning all potentially costly runtime operations into
 	// no-ops.
+	//
+	// TODO this should use atomic load/store to test disabled
+	// prior to taking the lock, do please.
 	disabled bool
 }
 
-func NewRecorder(opts Options) basictracer.SpanRecorder {
+func NewRecorder(opts Options) *Recorder {
 	if len(opts.AccessToken) == 0 {
-		// TODO maybe return a no-op recorder instead?
-		panic("LightStep Recorder options.AccessToken must not be empty")
+		fmt.Println("LightStep Recorder options.AccessToken must not be empty")
+		return nil // Note: this is a non-nil interface w/ a nil Recorder
```

Comment:
Somehow I'm not comforted by that comment. We can return `nil` here, but then I think we need to handle that case in the methods, e.g.
```go
func (r *Recorder) RecordSpan(raw basictracer.RawSpan) {
  if r == nil {
    return
  }
 ...
```